src/gui/raster/qgscolorrampshaderwidget.cpp

```diff
@@ -118,6 +120,18 @@ void QgsColorRampShaderWidget::setRasterDataProvider( QgsRasterDataProvider *dp
 void QgsColorRampShaderWidget::setRasterBand( int band )
 {
   mBand = band;
+  // Set the maximum number of digits in the precision spin box
+  if ( mRasterDataProvider )
+  {
+    const int maxDigits { QgsGuiUtils::significantDigits( mRasterDataProvider->dataType( mBand ) ) };
+    // Get current value
+    const int currentPrecision { mLabelPrecisionSpinBox->value() };
+    mLabelPrecisionSpinBox->setMaximum( maxDigits );
+    if ( currentPrecision > maxDigits )
+    {
+      mLabelPrecisionSpinBox->setValue( maxDigits );
+    }
```

Comment:
Is this needed? Won't setting the maximum for the spin box already handle this situation?

Reply:
I'm not sure: does it? I'll check. There is also the question if valueChanged is triggered (it should).